test.cpp

```diff
@@ -78,32 +78,41 @@ static void testcase(const std::string &name, void (*f)(), int argc, char * cons
 #define TEST_CASE(F)    (testcase(#F, F, argc, argv))
 
 
+static simplecpp::TokenList makeTokenList(const char code[], std::vector<std::string> &filenames, const std::string &filename=std::string(), simplecpp::OutputList *outputList=nullptr)
+{
+    std::istringstream istr(code);
+    return simplecpp::TokenList(istr,filenames,filename,outputList);
+}
+
+static simplecpp::TokenList makeTokenList(const char code[], std::size_t size, std::vector<std::string> &filenames, const std::string &filename=std::string(), simplecpp::OutputList *outputList=nullptr)
+{
+    std::istringstream istr(std::string(code, size));
+    return simplecpp::TokenList(istr,filenames,filename,outputList);
+}
 
-static std::string readfile(const char code[], int sz=-1, simplecpp::OutputList *outputList=nullptr)
+static simplecpp::TokenList makeTokenList(const char code[])
 {
-    std::istringstream istr(sz == -1 ? std::string(code) : std::string(code,sz));
     std::vector<std::string> files;
-    return simplecpp::TokenList(istr,files,std::string(),outputList).stringify();
+    return makeTokenList(code, files);
```

Comment:
this looks dangerous. the returned TokenList will reference the local `files`

Reply:
You are right. I didn't think about this.

We didn't detect this in Cppcheck - I have this enabled in my IDE and it did not report anything.

Reply:
I filed https://trac.cppcheck.net/ticket/10975. It is impossible to detect the issue I introduced though since we have no idea what is done with the reference.

Reply:
Done.